### PR TITLE
New error when a PackageReference item does not have a corresponding PackageVersion.

### PR DESCRIPTION
**Pull request by cristinamanum**
New error when a PackageReference item does not have a corresponding PackageVersion. Bug https://github.com/NuGet/Home/issues/9341

## Bug

Fixes: Bug https://github.com/NuGet/Home/issues/9341
Regression: No  
  

## Fix
Add a new validation and error.

## Testing/Validation

Tests Added: Yes
Validation:  Unit tests.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs [hunk 1]
```diff
@@ -420,6 +420,13 @@ private async Task<bool> AreCentralVersionRequirementsSatisfiedAsync()
 
                 return false;
             }
+            IEnumerable<LibraryDependency> packageReferencedDependenciesWithoutCentralVersionDefined = _request.Project.TargetFrameworks.SelectMany(tfm => tfm.Dependencies.Where(d=>d.LibraryRange.VersionRange == null));
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs]: Could probably use Any check on this whole thing without assigning a variable.

There's some room for formatting improvements too.

File: src/NuGet.Core/NuGet.Commands/Strings.resx [hunk 2]
```diff
@@ -1102,4 +1102,7 @@ For more information, visit https://docs.nuget.org/docs/reference/command-line-r
     <value>The project {0} is using CentralPackageVersionManagement, a NuGet preview feature.</value>
     <comment>{0} the project path.</comment>
   </data>
+  <data name="Error_CentralPackageVersions_MissingPackageVersion" xml:space="preserve">
+    <value>The PackageReference items {0} do not have corresponding PackageVersions.</value>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/Strings.resx]: The item name is `PackageVersion` right? 

We should not pluralize the MSBuild item names.

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs (hunk 1 above)
Comment (cristinamanum) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs]: Fixed formatting. The variable is used to get the package Ids that need to be fixed.

File: src/NuGet.Core/NuGet.Commands/Strings.resx (hunk 2 above)
Comment (cristinamanum) [reply src/NuGet.Core/NuGet.Commands/Strings.resx]: Removed the "s".

**Review by nkolev92 — APPROVED** [review]

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs [hunk 3]
```diff
@@ -420,6 +420,13 @@ private async Task<bool> AreCentralVersionRequirementsSatisfiedAsync()
 
                 return false;
             }
+            IEnumerable<LibraryDependency> packageReferencedDependenciesWithoutCentralVersionDefined = _request.Project.TargetFrameworks.SelectMany(tfm => tfm.Dependencies.Where(d => d.LibraryRange.VersionRange == null));
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs:423]: line of whitespace between closing brace of above code block and this new variable definition.

**Review by zivkan — APPROVED** [review]
This any your other PR that adds NU1011 are going to have merge conflicts, depending on which one gets merged first. Pretty much every file in the PR will have a merge conflict.

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs (hunk 3 above)
Comment (cristinamanum) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs:423]: I did the merge and got a successful CI run. Unfortunately I missed to change the space suggestion above. I will do it next time I will update the file.

**Pull request merged**